src/init_interp/initInterpMultilevelContainer.F90

```diff
@@ -576,14 +576,24 @@ subroutine get_levmaxurbgrnd_metadata(ncid_source, ncid_dest, &
        ! source and dest soil profiles differ despite having the same number of
        ! layers. The model will handle such cases without scaling
        ! the finidat variables that need scaling by soil layer thickness.
-       allocate(dzsoi_data_source(beg_dest:end_dest, nlev_source))
        do level = 1, nlev_source
           do g = beg_dest, end_dest
              dzsoi_data_source(g,level) = dzsoi_data_dest(g,level)
           end do
        end do
-    else  ! fail with helpful error message
-       call interp_levgrnd_check_source_file(ncid_source, dzsoi_varname)
+    else  ! different number of layers and .not. on_source
+       ! Rather than failing when DZSOI is not present on the finidat file,
+       ! construct dzsoi_data_source from coord_data_source
+       do g = beg_dest, end_dest
+          dzsoi_data_source(g,1) = 2._r8 * coord_data_source(g,1)
+          if (nlev_source > 1) then
+             do level = 2, nlev_source
+                dzsoi_data_source(g,level) = 2._r8 * &
+                   (coord_data_source(g,level) - &
+                    sum(dzsoi_data_source(g,1:level-1)))
```

Comment:
- [ ] Can you double-check this expression? I haven't worked it out carefully, but I think it should be `2*coord_data_source - sum(...)` – not `2*(coord_data_source - sum(...))`

Reply:
I think it's correct as is:
layer2node - sum(dz1) = dz2/2
layer3node - sum(dz1+dz2) = dz3/2
and so on.
But I could be missing something...

Reply:
Sorry, you're absolutely right. I got confused and thought we were calculating interface depths rather than thicknesses.

Reply:
Also... I hate to suggest this, but I could add a unit test to check this. Not sure if I can use the existing tests again as templates or if I need to start from scratch for this one...

Reply:
Up to you. The thought did occur to me. I'm not sure it's necessary in this case, given that this code is just here relatively short-term for backwards compatibility – as long as you have convinced yourself that it's correct (I'm reasonably convinced now, though haven't thought it through 100% carefully). If you do want to add tests, I'd suggest extracting this new little block to a subroutine in initInterpUtils, which has an associated test file.

I'll let you make the call depending on how much better it will let you sleep at night :-)  I'm fine either way.